tests/suites/test_suite_mpi.function

```diff
@@ -1720,6 +1720,378 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE */
+void mbedtls_mpi_core_add_if( char * input_l, char * input_r,
+                              char * input_X4, int carry4,
+                              char * input_X8, int carry8 )
+{
+    mbedtls_mpi X4, X8, l, r;
+    mbedtls_mpi_uint *la = NULL;
+    mbedtls_mpi_uint *ra = NULL;
+    mbedtls_mpi_uint *Xa = NULL;
+    mbedtls_mpi_uint *da = NULL;
+
+    mbedtls_mpi_init( &X4 );
+    mbedtls_mpi_init( &X8 );
+    mbedtls_mpi_init( &l );
+    mbedtls_mpi_init( &r );
+
+    TEST_EQUAL( mbedtls_test_read_mpi( &l, input_l ), 0 );
+    TEST_EQUAL( mbedtls_test_read_mpi( &r, input_r ), 0 );
+    TEST_EQUAL( mbedtls_test_read_mpi( &X4, input_X4 ), 0 );
+    TEST_EQUAL( mbedtls_test_read_mpi( &X8, input_X8 ), 0 );
+
+    /* We only need to work with one of (X4, carry4) or (X8, carry8) depending
+     * on sizeof(mbedtls_mpi_uint)
+     */
+    mbedtls_mpi *X = ( sizeof(mbedtls_mpi_uint) == 4 ) ? &X4 : &X8;
+    mbedtls_mpi_uint carry = ( sizeof(mbedtls_mpi_uint) == 4 ) ? carry4 : carry8;
+
+    /* All of the inputs are +ve (or zero) */
+    TEST_EQUAL( l.s, 1 );
+    TEST_EQUAL( r.s, 1 );
+    TEST_EQUAL( X->s, 1 );
+
+    /* Test cases are such that l <= r, so #limbs should be <= */
+    TEST_ASSERT( l.n <= r.n );
+    TEST_ASSERT( X->n <= r.n );
+
+    /* Now let's get arrays of mbedtls_mpi_uints, rather than MPI structures */
+    la = mbedtls_calloc( r.n, sizeof(mbedtls_mpi_uint) );
+    ra = mbedtls_calloc( r.n, sizeof(mbedtls_mpi_uint) );
+    Xa = mbedtls_calloc( r.n, sizeof(mbedtls_mpi_uint) );
+    da = mbedtls_calloc( r.n, sizeof(mbedtls_mpi_uint) );
+
+    TEST_ASSERT( la != NULL );
+    TEST_ASSERT( ra != NULL );
+    TEST_ASSERT( Xa != NULL );
+    TEST_ASSERT( da != NULL );
+
+    /* Populate the arrays. As the mbedtls_mpi_uint[]s in mbedtls_mpis (and as
+     * processed by mbedtls_mpi_core_add_if()) are little endian, we can just
+     * copy what we have as long as MSBs are 0 (which they are from calloc())
+     */
+    memcpy( la, l.p, l.n * sizeof(mbedtls_mpi_uint) );
+    memcpy( ra, r.p, r.n * sizeof(mbedtls_mpi_uint) );
+    memcpy( Xa, X->p, X->n * sizeof(mbedtls_mpi_uint) );
+
+    /* 1a) a += b, cond = 0 => there should be no carry */
```

Comment:
The comments and the executable code use different variable names. Please reconcile.

Reply:
This was the first test case I wrote. I agree it's a bit scrappy.